pkg/operation/botanist/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -603,6 +603,13 @@ func (d *deployer) deploy(ctx context.Context, operation string) (extensionsv1al
 		}
 
 		if ccdUnitContent != nil {
+			// We do not want to overwrite a valid Bootstraptoken with the tokenPlaceholder
+			for i, file := range downloaderFiles {
+				if file.Path == downloader.PathBootstrapToken {
+					downloaderFiles = append(downloaderFiles[:i], downloaderFiles[i+1:]...)
```

Comment:
Although, this works for removing one slice element, I find this code a bit confusing and is prone to errors when changed.
What about instead appending to `files` in this loop and simple not appending if its the bootstrap token path?

Reply:
fair enough I added your suggestion. WDYT?

Reply:
fair enough 👍🏼  I added your suggestion